.storybook/helpers/argTypes.js

```diff
@@ -23,6 +23,20 @@ function getDefaultValue(prop) {
       : cleanStrValue(def);
 }
 
+function filterPaddingProps(argTypes) {
+  const descript = '**Padding** *(accepts shorthands, eg. `p="sm xl lg"`)*. Also supports **quick props** **`px`**,  **`py`**, **`pt`**, **`pr`**, **`pb`**, **`pl`** *(eg. `px="sm" pt="lg"`)*';
+  const paddingProps = ['p', 'px', 'py', 'pt', 'pr', 'pb', 'pl'];
+
+  Object.keys(argTypes).forEach(prop => {
+    if (!paddingProps.includes(prop)) return;
+    if (prop === 'p') {
+      argTypes[prop].description = descript;
+    } else {
+      delete argTypes[prop];
+    }
+  });
```

Comment:
Not a blocker; it's always recommended to avoid mutation by reference and mutation on a `forEach` (when you have the other array methods):

```suggestion
  return Object.keys(argTypes).reduce((acc, prop) => {
    if (!paddingProps.includes(prop)) return acc;
    if (prop === 'p') {
      acc[prop].description = descript;
    } else {
      delete acc[prop];
    }
  }, { ...argTypes });
```

1. Instead of mutating the reference, it creates a shallow copy to work with.
2. It deletest on the accumulator directly.

Reply:
Agree. Though above code isn't working as expected:
![Screenshot 2021-03-24 at 09 00 34](https://user-images.githubusercontent.com/5729421/112275342-ce9c0a00-8c7f-11eb-9e7f-92e860c26cd8.jpg)
I've added a new commit where I just iterate through shallow copy's keys.

Reply:
Is it possible that `argTypes` is `undefined` on some point? because that error looks like it's being generated by `Object.keys` or the spread on the accumulator.